File: be/transfer.py
# 참고로 파이썬코는 다음과 같다.

from pymongo import MongoClient

# 로컬 MongoDB 연결 설정
local_client = MongoClient('mongodb://localhost:27017/')
local_db = local_client['hm-shopping3']
local_collection = local_db['orders']

# MongoDB Atlas 연결 설정
atlas_client = MongoClient(
    'mongodb+srv://idim7:bzeEcNaou6xGibHd@cluster0.dwajw45.mongodb.net/hm-shopping3')
atlas_db = atlas_client['hm-shopping3']
atlas_collection = atlas_db['orders']

# 로컬 컬렉션에서 모든 문서 가져오기
documents = local_collection.find()

# MongoDB Atlas 컬렉션에 문서 삽입
atlas_collection.insert_many(documents)

print("Documents copied successfully!")

File: be/transfer.js
const { MongoClient } = require('mongodb');

// 로컬 MongoDB 연결 설정
const localUrl = 'mongodb://localhost:27017/';
const localClient = new MongoClient(localUrl);
const localDb = localClient.db('hm-shopping3');
const localCollection = localDb.collection('orders');

// MongoDB Atlas 연결 설정
const atlasUrl = 'mongodb+srv://idim7:bzeEcNaou6xGibHd@cluster0.dwajw45.mongodb.net/hm-shopping3';
const atlasClient = new MongoClient(atlasUrl);
const atlasDb = atlasClient.db('hm-shopping3');
const atlasCollection = atlasDb.collection('orders');

const transferDbData=async()=> {
  try {
    await localClient.connect();
    await atlasClient.connect();

    const documents = await localCollection.find().toArray();

    if (documents.length > 0) {
      await atlasCollection.insertMany(documents);
      console.log("Documents copied successfully!");
    } else {
      console.log("No documents to copy.");
    }
  } catch (error) {
    console.error("Error during data transfer:", error);
  } finally {
    await localClient.close();
    await atlasClient.close();
  }
}

module.exports=transferDbData;